### PR TITLE
Toggle Line Comment - removing leading whitespace breaks Jade preprocessor #8012

### PR DESCRIPTION
**Pull request by mccricardo**
Fixing issue #8012 concerning toggle line comment for Jade files. 

Jade comments should be placed on the beginning and not on the beginning of the line.

Two tests were written to test comment and uncomment.

### CODE REVIEW

File: src/editor/EditorCommandHandlers.js [hunk 1]
```diff
@@ -237,6 +241,27 @@ define(function (require, exports, module) {
...
+    /**
+     * @private
+     * Given a line to be commented it will determine where the comment prefix position is.
+     * 
+     * @param {string} language id
+     * @param {string} line to be commented
+     * @return {integer}
+     */
+    function _getPrefixPos(language, line) {                        
+        if (language === "jade") {
```
Comment (peterflynn) [inline src/editor/EditorCommandHandlers.js]: EditorCommandHandlers shouldn't have hardcoded knowledge of specific languages -- this should be a general flag, e.g. `Language.indentLineComments`.  I'd imagine this same issue affects braceless/classic SASS too, for example.

File: src/editor/EditorCommandHandlers.js [hunk 2]
```diff
@@ -207,7 +209,9 @@ define(function (require, exports, module) {
         if (containsNotLineComment) {
             // Comment out - prepend the first prefix to each line
             for (i = startLine; i <= endLine; i++) {
-                editGroup.push({text: prefixes[0], start: {line: i, ch: 0}});
+                // For each line we need to determine the position where de comment prefix should be inserted
+                prefixPos = _getPrefixPos(language, doc.getLine(i));
```
Comment (peterflynn) [inline src/editor/EditorCommandHandlers.js]: The `language` var was removed from this function more recently, so you'll need to re-add it

File: src/editor/EditorCommandHandlers.js [hunk 3]
```diff
@@ -207,7 +209,9 @@ define(function (require, exports, module) {
...
+                editGroup.push({text: prefixes[0], start: {line: i, ch: prefixPos}});
             }
 
             // Make sure tracked selections include the prefix that was added at start of range
```
Comment (peterflynn) [inline src/editor/EditorCommandHandlers.js]: It looks like this code that deals with `trackedSel` contains assumptions about the comment being inserted at `ch: 0`.  You may need to update it to ensure it works as intended in the new case too.

**Comment by mccricardo** [comment]
Closing due to inactivity from my end.

**Pull request closed without merging**